### PR TITLE
Instrument api with opentelemetry

### PR DESCRIPTION
**Pull request by GabrielBarberini**
This PR intends to instrument infinity-API with opentelemetry hosting spans on Uptrace.

In terms of cost, [Uptrace provides 50GB or 2500 time-series each month for free](https://uptrace.dev/pricing), which probably will suit our needs for years.

## Summary by CodeRabbit

- **New Features**
  - Introduced OpenTelemetry tracing for various API routes, enhancing monitoring and performance tracking.
  - Added a configuration file for Gunicorn to improve deployment settings.

- **Updates**
  - Updated project version to "1.2.0" and revised description in project metadata.
  - Updated API version to "1.2.0 BETA".

- **Dependencies**
  - Added new dependencies: `uptrace`, `opentelemetry.instrumentation.fastapi`, `opentelemetry-api`, and `opentelemetry-sdk`.

- **Bug Fixes**
  - Adjusted import statement formatting to comply with code quality standards.

### CODE REVIEW

File: lib/settings/gunicorn.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import uptrace
+from opentelemetry.instrumentation.fastapi import FastAPIInstrumentor
+from lib.secrets import Secrets
+
+
+def post_fork(server, worker):  # pylint: disable=unused-argument
+    uptrace.configure_opentelemetry(
+        dsn=Secrets.get_secret("UPTRACE_DSN"),
+        service_name="infinity-api",
+        service_version="1.2.0",
```
Comment (luimot) [inline lib/settings/gunicorn.py:10]: There's a lot of places we're writing the API version, can we automate it somehow?

**Review by luimot — APPROVED** [review]
Didn't try it out but seen it in action outside of GH, seems simple so don't have any worries, seems good, good start for opentelemetry on our API

File: lib/settings/gunicorn.py (hunk 1 above)
Comment (GabrielBarberini) [reply lib/settings/gunicorn.py:10]: Good point, I'll think in a way to have a single source of truth for that

**Comment by GabrielBarberini** [comment]
@coderabbitai review

**Pull request merged**